### ISSUE TITLE
Bug: Pagination `getFirstPageNumber()` & `getLastPageNumber()` returns unexpected result.

### ISSUE THREAD

**Issue by sammyskills** [codeigniter4/CodeIgniter4#8254]
### PHP Version

8.2

### CodeIgniter4 Version

4.4.3

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

apache

### Database

MariaDB 10.4.18

### What happened?

Within the view file for pagination links, the `getFirstPageNumber()` & `getLastPageNumber()` methods don't seem to return the expected results. From the [docs](https://codeigniter4.github.io/CodeIgniter4/libraries/pagination.html#getfirstpagenumber-getlastpagenumber):
> These methods return page numbers to the first and last pages in the result set.

Let's say your `$pager->links()` prints out the following:
```text
1 | 2 | 3 | 4 | 5
```
calling the `getFirstPageNumber()` & `getLastPageNumber()` returns `1` and `5` respectively. 

And when you navigate to the next page, with links as follows:
```text
3 | 4 | 5 | 6 | 7
```
`getFirstPageNumber()` & `getLastPageNumber()` returns `3` and `7` respectively.

### Steps to Reproduce

Follow the docs and create a simple pagination.

### Expected Output

I expected the code to work differently, `getFirstPageNumber()` should return `1` & `getLastPageNumber()` should return the count of the total number of pages, irrespective of the page being viewed.  As seen in the source code: https://github.com/codeigniter4/CodeIgniter4/blob/14b4350b3685d33018ce7ba0c7d0a368f25962ce/system/Pager/PagerRenderer.php#L386-L389
https://github.com/codeigniter4/CodeIgniter4/blob/14b4350b3685d33018ce7ba0c7d0a368f25962ce/system/Pager/PagerRenderer.php#L402-L405
https://github.com/codeigniter4/CodeIgniter4/blob/14b4350b3685d33018ce7ba0c7d0a368f25962ce/system/Pager/PagerRenderer.php#L86-L89

**Comment by kenjis** [comment]
What does this issue mean?
Do you mean you want the current behavior to change?

**Comment by sammyskills** [comment]
Well, if the current behaviour is the intended, then I think it is misleading (IMO).

How do devs get the first page number and last page number irrespective of the result?

**Comment by kenjis** [comment]
> How do devs get the first page number and last page number irrespective of the result?

The first page number is always 1.
and use getPageCount().

**Comment by sammyskills** [comment]
Then is there an actual use case for the current behaviour of `getFirstPageNumber()` & `getLastPageNumber()`?

**Comment by kenjis** [comment]
E.g., changing the decoration of the first and last page numbers?

**Comment by kenjis** [comment]
How about this?
```diff
--- a/system/Pager/PagerRenderer.php
+++ b/system/Pager/PagerRenderer.php
@@ -25,14 +25,14 @@ use CodeIgniter\HTTP\URI;
 class PagerRenderer
 {
     /**
-     * First page number.
+     * First page number in the current page set being displayed.
      *
      * @var int
      */
     protected $first;
 
     /**
-     * Last page number.
+     * Last page number in the current page set being displayed.
      *
      * @var int
      */
@@ -85,8 +85,10 @@ class PagerRenderer
      */
     public function __construct(array $details)
     {
+        // `first` and `last` will be updated by `setSurroundCount()`.
         $this->first        = 1;
         $this->last         = $details['pageCount'];
+
         $this->current      = $details['currentPage'];
         $this->total        = $details['total'];
         $this->uri          = $details['uri'];
```

**Comment by sammyskills** [comment]
> E.g., changing the decoration of the first and last page numbers?

Well, okay, but are those methods not supposed to be in sync with the `getFirst()` and `getLast()` methods?

For example, if the `getFirst()` always returns the url of the first result in the entire dataset, why does `getFirstPageNumber()` behave otherwise?

**Comment by kenjis** [comment]
> For example, if the getFirst() always returns the url of the first result in the entire dataset, why does getFirstPageNumber() behave otherwise?

I agree that it is difficult to understand, but it is probably because of historical reasons.

**Comment by kenjis** [comment]
There are many different methods of Pagination, and they are indeed confusing, but if we change the method name, it is a breaking change. So it is not easy to change.

**Comment by sammyskills** [comment]
🤔

How about factoring it in v4.5 ?

**Comment by sammyskills** [comment]
Or, the docs could be updated for better understanding, including the suggested change in https://github.com/codeigniter4/CodeIgniter4/issues/8254#issuecomment-1826435461

**Comment by kenjis** [comment]
If you can improve Pagination without breaking existing apps by adding new, easy-to-understand methods, please send PR to 4.5.
Otherwise, I don't see the value in going to the trouble of breaking existing apps to make the change.
In fact, I think the names are just confusing, but they work.

The documentation has room for improvement.

**Comment by kenjis** [comment]
I sent PR #8256
